raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -32,109 +28,28 @@ def decode_event(abi: Dict, log: Dict):
     return get_event_data(event_abi, log)
 
 
-def encode_function_call(abi: Dict, function: str, args: List=list()):
-    fn_abi = find_matching_fn_abi(abi, function, args)
-    fn_types = get_abi_input_types(fn_abi)
-    return encode_abi(fn_types, args)
-
-
 class ContractProxy:
-    """ Proxy to interact with a smart contract through the rpc interface. """
-
     def __init__(
-            self,
-            sender: Address,
-            abi: Dict,
-            contract_address: Address,
-            call_function: Callable,
-            transact_function: Callable,
-            estimate_function: Optional[Callable] = None):
-
-        sender = to_canonical_address(sender)
-        contract_address = to_canonical_address(contract_address)
-
-        self.abi = abi
-        self.call_function = call_function
-        self.contract_address = contract_address
-        self.estimate_function = estimate_function
-        self.sender = sender
-        self.transaction_function = transact_function
-        self.valid_kargs = {'gasprice', 'startgas', 'value'}
-
-    def _check_function_name_and_kargs(self, function_name: str, kargs):
-        functions = [x['name'] for x in self.abi if x['type'] == 'function']
-        if function_name not in functions:
-            raise InvalidFunctionName('Unknown function {}'.format(function_name))
-
-        invalid_args = set(kargs.keys()).difference(self.valid_kargs)
-        if invalid_args:
-            raise TypeError('got an unexpected keyword argument: {}'.format(
-                ', '.join(invalid_args),
-            ))
+        self,
+        jsonrpc_client,
+        contract=None
+    ):
+        self.jsonrpc_client = jsonrpc_client
+        self.contract = contract
 
     def transact(self, function_name: str, *args, **kargs):
-        self._check_function_name_and_kargs(function_name, kargs)
-        data = self.get_transaction_data(self.abi, function_name, args)
+        assert self.contract is not None
+        data = ContractProxy.get_transaction_data(self.contract.abi, function_name, args)
```

Comment:
This is a bit late, but why have `get_transaction_data` as a `staticmethod` instead of a free standing function?

Reply:
no other code uses it and it doesn't change state of the object; then it makes more sense keeping it coupled to the class.